### ISSUE TITLE
Template `LookupTable` class to support different numeric types (And other mappers if needed)

### ISSUE THREAD

**Issue by jabrap1** [macformula/racecar#285]
Template `LookupTable` Class and others in the mappers folder if they are not up to date

Description:
The `LookupTable` class is currently designed to work only with `float` types. To increase its flexibility, we could template the class to support different numeric types (e.g., `double`, `int`).

Proposal:
Add a template parameter `typename T` to `LookupTable`.

**Comment by BlakeFreer** [comment]
I agree with the templating but only for double / float. It doesn't make much sense to have an interpolating table when you're restricted to a discrete set.

**Comment by BlakeFreer** [comment]
This is a textbook use for concepts & `requires`, see `std::floating_point`
